pkg/info/info.go

```diff
@@ -55,5 +57,33 @@ var (
 	// This will be set via Goreleaser during the build process
 	Version string
 
+	// ReadBuildInfo is a function that read the build info from go build.
+	ReadBuildInfo = debug.ReadBuildInfo
+
 	UpstreamREADMETagged = fmt.Sprintf(UpstreamREADMETemplate, Version)
 )
+
+type InfoService interface {
+	// get the binary version of the current build from available sources
+	GetBuildVersion() string
+}
+
+type DefaultInfoServiceImpl struct {
+}
+
+func (i *DefaultInfoServiceImpl) GetBuildVersion() string {
+	// If the Version is set by Goreleaser, return it directly.
+	if len(Version) > 0 {
```

Comment:
Nice to use `Version != ""` since Version is `string` type, but the same logic and small performance improvement

Reply:
Thanks! Changed to != "" in the new commit.